x/liquidity/spec/03_state_transitions.md

```diff
@@ -1,174 +1,235 @@
-<!--
-order: 3
--->
+<!-- order: 3 -->
 
-# State Transitions
+ # State Transitions
+
+Messages (Msg) trigger state transitions.
 
 ## Coin Escrow for Liquidity Module Messages
 
-Three messages on the liquidity module need prior coin escrow before confirmation, which causes state transition on `Bank` module. Below lists are describing coin escrow processes for each given message type.
+Transaction confirmation causes state transition on `Bank` module. Some messages on the Liquidity Module require coin escrow before confirmation.
+
+The coin escrow processes for each message type are:
 
 ### MsgDepositWithinBatch
 
-To deposit coins into existing `Pool`, the depositor needs to escrow `DepositCoins` into `LiquidityModuleEscrowAccount`.
+To deposit coins into an existing `Pool`, the depositor must escrow `DepositCoins` into `LiquidityModuleEscrowAccount`.
 
 ### MsgWithdrawWithinBatch
 
-To withdraw coins from `Pool`, the withdrawer needs to escrow `PoolCoin` into `LiquidityModuleEscrowAccount`.
+To withdraw coins from a `Pool`, the withdrawer must escrow `PoolCoin` into `LiquidityModuleEscrowAccount`.
 
 ### MsgSwapWithinBatch
 
-To request coin swap, swap requestor needs to escrow `OfferCoin` into `LiquidityModuleEscrowAccount`.
+To request a coin swap, the swap requestor must escrow `OfferCoin` into `LiquidityModuleEscrowAccount`.
 
 ## LiquidityPoolBatch Execution
 
-Batch execution causes state transitions on `Bank` module. Below categories describes state transition executed by each process in `PoolBatch` execution.
+Batch execution causes state transitions on the `Bank` module. The following categories describe state transition executed by each process in the `PoolBatch` execution.
 
 ### Coin Swap
 
-After successful coin swap, coins accumulated in `LiquidityModuleEscrowAccount` for coin swaps are sent to other swap requestors(self-swap) or to the `Pool`(pool-swap). Also fees are sent to the `Pool`.
+After a successful coin swap, coins accumulated in `LiquidityModuleEscrowAccount` for coin swaps are sent to other swap requestors(self-swap) or to the `Pool`(pool-swap). Fees are also sent to the liquidity `Pool`.
+
+### LiquidityPool Deposit
+
+After a successful deposit transaction, escrowed coins are sent to the `ReserveAccount` of the targeted `Pool` and new pool coins are minted and sent to the depositor.
+
+### LiquidityPool Withdrawal
 
-### LiquidityPool Deposit and Withdraw
+After a successful withdraw transaction, escrowed pool coins are burned and a corresponding amount of reserve coins are sent to the withdrawer from the liquidity `Pool`.
 
-For deposit, after successful deposit, escrowed coins are sent to the `ReserveAccount` of targeted `Pool`, and new pool coins are minted and sent to the depositor.
+## Pseudo Algorithm for LiquidityPoolBatch Execution
```

Comment:
okay, here is where I start to wonder what the content purpose and structure are (help me, I need a lot of context, please and thank you!)

why are we mentioning the pseudo algo here? why do we give a link tot he Python file?

Reply:
The pseudo algo is there as reference for those who may feel curious about how this module is built from the ground.